src/main/java/com/visenze/datatracking/VisenzeAnalytics.java

```diff
@@ -0,0 +1,94 @@
+package com.visenze.datatracking;
+
+
+
+import android.content.Context;
+
+import com.visenze.datatracking.data.DataCollection;
+import com.visenze.datatracking.utils.Utils;
+
+import java.util.Date;
+import java.util.UUID;
+
+
+
+public class VisenzeAnalytics {
+
+    public static final String TAG = "DataTracking";
+    public static final long SESSION_TIMEOUT = 1800000; //30 min
+
+    private static VisenzeAnalytics mInstance;
+
+
+    private String uid;
+    private long sid;
+
+    private DataCollection dataCollection;
+
+
+
+    private VisenzeAnalytics(Context context, String uid) {
+        if (uid == null || uid.isEmpty()) {
+            uid = generateUid(context);
+        }
+        initDataTracking(context, uid);
+    }
+
+    private void initDataTracking(Context context, String uid) {
+        this.uid = uid;
+        this.sid = getSid(context);
+        dataCollection = new DataCollection(context);
+    }
+
+
+
+    public static synchronized VisenzeAnalytics getInstance(Context context) {
```

Comment:
this should be return getInstance(context, null);

Reply:
changed